modules/registrars/openprovider/Controllers/Hooks/DomainLockingEnabledController.php

```diff
@@ -0,0 +1,154 @@
+    {
+        $this->apiHelper = $apiHelper;
+        $this->domain = $domain;
+    }
+
+    public function handleDomainLockingClientArea($vars)
+    {
+        $id = $vars['domain']->id ?? null;
+
+        if (!$id) {
+            return "";
+        }
+        // Get the domain details
+        $domain = $this->domain->find($id);
+
+        // Check if OpenProvider is the provider
+        if (!$domain || $domain->registrar !== "openprovider") {
+            return "";
+        }
+
+        if ($this->isDomainLockingEnabled($domain)) {
+            return "";
+        }
+
+        return '
+                <script>
+                document.addEventListener("DOMContentLoaded", function() {
+                    var tabReglockListItem = document.querySelector(\'a[href="#tabReglock"]\');
+                    console.log(document.querySelector(\'a[href="#tabReglock"]\').parentElement);
+                    if (tabReglockListItem) {
+                        tabReglockListItem.parentElement.style.display = \'none\';
+                    }
+                });
+                document.addEventListener("DOMContentLoaded", function() {
+                    var alerts = document.querySelectorAll(\'.alert.alert-danger\');
+                    alerts.forEach(function(alert) {
+                        var strongTag = alert.querySelector(\'strong\');
+                        if (strongTag && strongTag.textContent.trim() === \'Domain Currently Unlocked!\') {
+                            alert.remove();
+                        }
+                    });
+                });
+            </script>';
+    }
+
+    public function handleDomainLockingClientSidebar(\WHMCS\View\Menu\Item $primarySidebar)
+    {
+        $id = $_REQUEST['id'] ?? ($_REQUEST['domainid'] ?? null);
+        if (!$id) {
+            return "";
+        }
+        // Get the domain details
+        $domain = $this->domain->find($id);
+        // Check if OpenProvider is the provider
+        if (!$domain || $domain->registrar !== "openprovider") {
+            return "";
+        }
+
+        if ($this->isDomainLockingEnabled($domain)) {
+            return "";
+        }
+
+        if (!is_null($primarySidebar->getChild('Domain Details Management'))) {
+            $primarySidebar->getChild('Domain Details Management')->removeChild('Registrar Lock Status');
+        }
+    }
+
+    public function handleDomainLockingAdminArea()
+    {
+        if (basename($_SERVER['PHP_SELF'] ?? (($_SERVER['SCRIPT_NAME'] ?? $_SERVER['SCRIPT_FILENAME']) ?? '')) !== "clientsdomains.php") {
+            return "";
+        }
+        $id = $_REQUEST['id'] ?? ($_REQUEST['domainid'] ?? null);
+        if (!$id) {
+            return "";
+        }
+        // Get the domain details
+        $domain = $this->domain->find($id);
+        // Check if OpenProvider is the provider
+        if (!$domain || $domain->registrar !== "openprovider") {
```

Comment:
may be better to move to constant `openprovider`

Reply:
In the repo we don't have a separate constant file to store such values. There for improved the code by using a class level constant value for "openprovider" since it is used in multiple places in the same file.

Reply:
okay got it